metricflow-semantics/metricflow_semantics/model/semantics/linkable_spec_resolver.py

```diff
@@ -463,15 +466,20 @@ def _get_metric_time_elements(self, measure_reference: Optional[MeasureReference
                 semantic_model=measure_semantic_model,
                 time_dimension_reference=measure_agg_time_dimension_reference,
             )
+            possible_metric_time_granularities = tuple(
+                time_granularity
+                for time_granularity in TimeGranularity
+                if defined_granularity.is_smaller_than_or_equal(time_granularity)
+            )
         else:
-            defined_granularity = DEFAULT_TIME_GRANULARITY
-
-        # It's possible to aggregate measures to coarser time granularities (except with cumulative metrics).
-        possible_metric_time_granularities = tuple(
-            time_granularity
-            for time_granularity in TimeGranularity
-            if defined_granularity.is_smaller_than_or_equal(time_granularity)
-        )
+            # If querying metric_time without metrics, will query from time spines.
+            # Defaults to DAY granularity if available in time spines, else smallest available granularity.
+            min_time_spine_granularity = min(self._time_spine_sources.keys())
```

Comment:
This default is because the time_spine_sources automatically include the DAY granularity (via `TimeSpineSource.create_from_manifest()`) if no custom time spines are found, correct?

Reply:
This default is set so that if you query ambiguous `metric_time` without metrics, we don't return nanoseconds if you have that time spine configured. Since `metric_time` defaults to `DAY` in all other scenarios (unless overridden by the metric's default granularity), this seemed like the expected behavior.